apps/tenant-management-webapp/src/app/pages/admin/services/pdf/testGenerate/generatePDF.tsx

```diff
@@ -0,0 +1,121 @@
+import { PdfGenerationPayload, PdfTemplate } from '@store/pdf/model';
+import { useSelector } from 'react-redux';
+import { RootState } from '@store/index';
+import styled from 'styled-components';
+import { GoAElementLoader } from '@abgov/react-components';
+
+interface GeneratePDFModalProps {
+  template: PdfTemplate;
+  onSave: (definition: PdfGenerationPayload) => void;
+}
+export const GeneratePDF: FunctionComponent<GeneratePDFModalProps> = ({ template, onSave }) => {
+  const [pdfPayload, setPdfPayload] = useState<PdfGenerationPayload>({
+    templateId: template.id,
+    data: { testVariable: 'some data' },
+    fileName: `${template.id}_${new Date().toJSON().slice(0, 19).replace(/:/g, '-')}.pdf`,
+  });
+  const indicator = useSelector((state: RootState) => {
+    return state?.session?.indicator;
+  });
+  const [errors, setErrors] = useState<Record<string, string>>({});
+
+  const hasFormErrors = () => {
+    return Object.keys(errors).length !== 0;
+  };
+  return (
+    <>
+      <GoAForm>
+        <GoAFormItem>
+          <EditorStyles>
+            <Editor
+              data-testid="form-schema"
+              height={150}
+              value={JSON.stringify(pdfPayload?.data, null, 2)}
+              onChange={(value) => setPdfPayload({ ...pdfPayload, data: JSON.parse(value) })}
+              language="json"
+              options={{
+                automaticLayout: true,
+                scrollBeyondLastLine: false,
+                tabSize: 2,
+                minimap: { enabled: false },
+              }}
+            />
+          </EditorStyles>
+        </GoAFormItem>
+      </GoAForm>
+
+      <FlexRow>
+        <PaddingRight>
+          <GoAButton
+            disabled={!pdfPayload?.templateId || indicator.show}
+            type="primary"
+            data-testid="form-save"
+            onClick={() => {
+              // if no errors in the form then save the definition
+              if (!hasFormErrors()) {
+                onSave(pdfPayload);
+              } else {
+                return;
+              }
+            }}
+          >
+            <GenerateButtonPadding>
+              Generate PDF
+              {indicator.show ? (
+                <SpinnerPadding>
+                  <GoAElementLoader visible={true} size="default" baseColour="#c8eef9" spinnerColour="#0070c4" />
+                </SpinnerPadding>
+              ) : (
+                <SpinnerSpace></SpinnerSpace>
+              )}
+            </GenerateButtonPadding>
+          </GoAButton>
+        </PaddingRight>
+
+        <GoAButton
+          data-testid="form-cancel"
+          type="secondary"
+          onClick={() => {
+            setPdfPayload({ templateId: null, data: { testVariable: 'some data' }, fileName: null });
+            setErrors({});
+          }}
+        >
+          Reset PDF
+        </GoAButton>
+      </FlexRow>
+    </>
+  );
+};
+
+const EditorStyles = styled.div`
```

Comment:
please move these to its style-components file

Reply:
Moved, Thanks